### ISSUE TITLE
Custom Deploy Strategy - Cannot redeclare class

### ISSUE THREAD

**Issue by errodr** [rocketeers/rocketeer#439]
I am trying to create a custom deploy strategy. To that end I have created a strategies directory in my .rocketeer folder. In that directory I have created a class:

```
<?php
namespace Test;

use Rocketeer\Abstracts\Strategies\AbstractStrategy;
use Rocketeer\Bash;
use Rocketeer\Interfaces\Strategies\DeployStrategyInterface;

class CustomStrategy extends AbstractStrategy implements DeployStrategyInterface
{
    protected $description = 'Testing custom deploy strategy.';

    public function deploy($destination = null)
    {
        return true;
    }

    public function update($reset = true)
    {
        return true;
    }
}
```

In my .rocketeer/strategies.php file I have the following configuration:

```
<?php
use Rocketeer\Binaries\PackageManagers\Composer;
use Rocketeer\Tasks\Subtasks\Primer;

return array(

    // Task strategies
    //
    // Here you can configure in a modular way which tasks to use to
    // execute various core parts of your deployment's flow
    //////////////////////////////////////////////////////////////////////

    // Which strategy to use to check the server
    'check'        => '',

    // Which strategy to use to create a new release
    'deploy'       => 'Test\Custom',
```

When I run a rocketeer command, i.e. "rocketeer strategies" or "rocketeer deploy --pretend" I get the following error:

```
PHP Fatal error:  Cannot redeclare class Test\CustomStrategy in .rocketeer/strategies/Custom.php on line 8
```

Is this a bug? I don't see how the class is being redeclared.

**Comment by ShonM** [comment]
#409 may help you!

**Comment by errodr** [comment]
Thanks. I saw that issue but my classes are located in the .rocketeer/strategies folder so it seems that this may be another issue.

Also, how are you getting the PHP Stack trace seen in your post?

**Comment by ShonM** [comment]
The issue and solution are the same: Move your own classes outside of `.rocketeer`. Not sure about traces, those always show on everything for me. PHP 5.6, xdebug, Rocketeer via Composer (not Phar).

**Comment by Anahkiasen** [comment]
Actually that's a bug it should ignore the strategies subfolder when computing hash :p will push a fix

**Comment by errodr** [comment]
I'm confused...

In http://rocketeer.autopergamene.eu/#/docs/docs/II-Concepts/Strategies

It says to put items in the .rocketeer/strategies subfolder...

**Comment by errodr** [comment]
Great, thanks...

**Comment by ShonM** [comment]
My mistake

**Comment by Anahkiasen** [comment]
Is this 2.0 or 3.0-dev though?

**Comment by errodr** [comment]
Actually I get the error on both 2.1.0 and 3.0-dev

**Comment by Anahkiasen** [comment]
Should be fixed on 3.0-dev

**Comment by errodr** [comment]
My mistake, you are correct.
